src/components/Badge/index.tsx

```diff
@@ -1,166 +1,152 @@
-  `,
-  rounded: ({ theme: { space, radii } }: { theme: Theme }) => css`
-    border-radius: ${radii.large};
-    font-size: 10px;
-    height: ${space['2']};
-    padding: ${space['0.25']} ${space['0.75']};
-    text-transform: uppercase;
-  `,
-  small: ({ theme: { space } }: { theme: Theme }) => css`
-    font-size: 12px;
-    line-height: ${space['3']};
-    height: ${space['3']};
-  `,
-  xsmall: ({ theme: { space, radii } }: { theme: Theme }) => css`
-    border-radius: ${radii.default};
-    font-size: 12px;
-    line-height: ${space['2.25']};
-    height: ${space['2.25']};
-  `,
-  xxsmall: ({ theme: { space, radii } }: { theme: Theme }) => css`
-    border-radius: ${radii.default};
-    font-size: 10px;
-    line-height: ${space['2']};
-    height: ${space['2']};
-    padding: 0 ${space['0.75']};
-  `,
+// TODO: remove when typography has been created
+const TEXT_SIZES = {
+  large: 14,
+  medium: 12,
+  small: 10,
 }
 
-type Variants = keyof typeof variants
-type Sizes = keyof typeof sizes
+export const SIZES = {
+  large: 32,
+  medium: 24,
+  small: 16,
+}
 
-export const badgeVariants = Object.keys(variants) as Variants[]
-export const badgeSizes = Object.keys(sizes) as Sizes[]
+export const PROMINENCES = {
+  default: 'default',
+  strong: 'strong',
+}
 
-const sizesStyle = ({ size, ...props }: { size?: Sizes; theme: Theme }) =>
-  sizes[size as Sizes]?.(props)
-const variantsStyle = ({
-  variant,
-  ...props
+/**
+ * Generate all styles available for badge based on prominence and variants
+ * @param prominence
+ * @param theme
+ */
+const generateStyles = ({
+  prominence,
+  theme,
 }: {
-  variant?: Variants
+  prominence: string
   theme: Theme
-}) => variants[variant as Variants]?.(props)
+}): Record<string, string> => {
+  const definedProminence =
+    prominence === PROMINENCES.strong ? capitalize(PROMINENCES.strong) : ''
+
+  const text = `text${definedProminence}` as keyof typeof theme.colors.primary
+  const background =
+    `background${definedProminence}` as keyof typeof theme.colors.primary
+
+  return {
+    ...SENTIMENTS.reduce(
+      (reducer, sentiment) => ({
+        ...reducer,
+        [sentiment]: `
+      color: ${theme.colors[sentiment][text]};
+      background: ${theme.colors[sentiment][background]}
+    `,
+      }),
+      {},
+    ),
+    disabled: `
+      color: ${theme.colors.neutral.textWeak};
+      background: ${theme.colors.neutral.backgroundStrong}
+    `,
+    neutral: `
+      color: ${
+        prominence === PROMINENCES.strong
+          ? theme.colors.neutral[text]
+          : theme.colors.neutral.textWeak
+      };
+      background: ${theme.colors.neutral[background]}
+    `,
+  }
+}
 
 type BadgeProps = {
-  variant?: Variants
-  size?: Sizes
-} & BoxProps
+  variant?: Color
+  size?: keyof typeof SIZES
+  prominence?: keyof typeof PROMINENCES
+  icon?: IconName
+  disabled?: boolean
+  className?: string
+}
 
-const StyledBox = styled(Box, {
+const StyledBox = styled('div', {
```

Comment:
There is a lot of Badge that still use xstyled props no ?

Reply:
Yes this PR is made to have breaking changes, I will migrate all badge to new interface